cloudbuild/python-scripts/insert_dynamic_partitions.py

```diff
@@ -0,0 +1,141 @@
+        f' {datetime.datetime.now()}'
+    )
+    # This is a buffer time to allow the read streams to be formed.
+    # Allows conflicting conditions by making sure that
+    # new (to be generated) partitions are not part of the current read stream.
+    time.sleep(2.5 * 60)
+
+
+def main(argv: Sequence[str]) -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--refresh_interval',
+        dest='refresh_interval',
+        help='Minutes between checking new data',
+        type=int,
+        required=True,
+    )
+    parser.add_argument(
+        '--project_name',
+        dest='project_name',
+        help='Project Id which contains the table to be read.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--dataset_name',
+        dest='dataset_name',
+        help='Dataset Name which contains the table to be read.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--table_name',
+        dest='table_name',
+        help='Table Name of the table which is read in the test.',
+        type=str,
+        required=True,
+    )
+
+    args = parser.parse_args(argv[1:])
+
+    # Providing the values.
+    project_name = args.project_name
+    dataset_name = args.dataset_name
+    table_name = args.table_name
+
+    execution_timestamp = datetime.datetime.now(tz=datetime.timezone.utc).replace(
+        hour=0, minute=0, second=0, microsecond=0
+    )
+    refresh_interval = int(args.refresh_interval)
+
+    # Set the partitioned table.
+    table_id = f'{project_name}.{dataset_name}.{table_name}'
+
+    # Now add the partitions to the table.
+    # Hardcoded schema. Needs to be same as that in the pre-created table.
+    simple_avro_schema_fields_string = (
+        '"fields": [{"name": "name", "type": "string"},{"name": "number",'
+        '"type": "long"},{"name" : "ts", "type" : {"type" :'
+        '"long","logicalType": "timestamp-micros"}}]'
+    )
+    simple_avro_schema_string = (
+        '{"namespace": "project.dataset","type": "record","name":'
+        ' "table","doc": "Avro Schema for project.dataset.table",'
+        f'{simple_avro_schema_fields_string}'
+        '}'
+    )
+
+    # hardcoded for e2e test.
+    # partitions[i] * number_of_rows_per_partition are inserted per phase.
+    partitions = [2, 1, 2]
+    # Insert 10000 - 30000 rows per partition.
+    # So, in a read up to 60000 new rows are read.
+    number_of_rows_per_partition = random.randint(1, 3) * 10000
+    #  BQ rate limit is exceeded due to large number of rows.
+    number_of_threads = 2
+    number_of_rows_per_thread = int(
+        number_of_rows_per_partition / number_of_threads
+    )
+
+    avro_file_local = 'mockData.avro'
+    table_creation_utils = utils.TableCreationUtils(
+        simple_avro_schema_string,
+        number_of_rows_per_thread,
+        table_id,
+    )
+
+    # Insert in phases.
```

Comment:
`# Insert iteratively.`
Iteration is more appropriate than phase. Please apply that to other comments in this loop.

Reply:
Fixed.